src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs

```diff
@@ -91,7 +91,12 @@ public static async Task<RemoteHostClient> CreateAsync(HostWorkspaceServices ser
                     new object?[] { uiCultureLCID, cultureLCID },
                     cancellationToken).ConfigureAwait(false);
 
-                client.Started();
+                var processId = await client._endPoint.InvokeAsync<int>(
+                       nameof(IRemoteHostService.GetProcessIdAsync),
+                       Array.Empty<object?>(),
+                       cancellationToken).ConfigureAwait(false);
```

Comment:
Why another RPC call? You can return the process id from InitializeGlobalState.

Reply:
Alternatively, you can keep GetProcessIdAsync separate and only call it from VisualStudioRemoteHostClientProvider.CreateHostClientAsync if debugger is attached.

---
In reply to: [475823792](https://github.com/dotnet/roslyn/pull/47053#discussion_r475823792) [](ancestors = 475823792)

Reply:
I was worried initially, but then I remembered that we make many thousands of calls regularly so even a small clarity improvement is likely worth the cost of separation.